### ISSUE TITLE
SC2046 - Quote this to prevent word splitting

### ISSUE THREAD

**Issue by erwinkramer** [koalaman/shellcheck#648]
![image](https://cloud.githubusercontent.com/assets/4255748/14420056/1c962620-ffca-11e5-8ef4-c140bd650bf2.png)

I quoted and it still tells me i didn't

**Comment by srstevenson** [comment]
Quote the command substitution:
```sh
kill -9 "$(jobs -p)"
```

**Comment by erwinkramer** [comment]
Oh right. Thanks!

**Comment by erwinkramer** [comment]
Doing it like srstevenson shows gives me something like: 
2524: arguments must be process or job IDs

**Comment by vutny** [comment]
@erwinkramer In your particular case shellcheck gives a false positive.
Generally any command substitutions should be quoted, but here you actually *need a word splitting*. That's because `kill` command recognizes a list of PIDs, not a single string with space separated numbers.

Just remove quotes and disable the warning like this:
```
# shellcheck disable=SC2046
kill -9 $(jobs -p)
```
Remember that such command will also throw a syntax error if there are no background jobs in your shell session. I humbly suggest to redirect `stderr` to `/dev/null`: `kill -9 $(jobs -p) 2>/dev/null`.

**Comment by koalaman** [comment]
Huh, the arrow is off by four. Something's wrong with the tab handling

**Comment by koalaman** [comment]
Arrow now points to the right thing.

The right thing in this case would be to disable to warning, as in this case word splitting is harmless and desirable.
